docs/ocis/adr/0021-service-authorization.md

```diff
@@ -0,0 +1,127 @@
+An application could be the search service (or only a content extraction service), a workflow service, a publication service or any other service.
+
+#### Q&A
+
+* Which services need a service account?
+Every service. Either on behalf of a logged in user (thumbnails or graph), or without a user being logged in (antivirus, search indexing). The MS graph api calls the former *delegated permissions* and the latter *application permissions*. See the [MS Graph permissions overview](https://learn.microsoft.com/en-us/graph/permissions-overview?tabs=http) 
+
+Service | Delegated Permissions | Application Permissions | Comment
+-|-|-
+graph | `Files.ReadWrite.All`, `Drive.Read` | | The graph translates all requests of the currently logged in user into CS3 requests. For admins and space admins `Drives.ReadWrite.All` would allow managing drives.
+search | `Files.Read.All` | | Search needs to be able to read the indexed content, which is equivalent to all files, the currently logged in user has access to
+indexer | | `Files.Read.All` | The indexing process is triggered asynchronously and the admin needs to consent on behalf of all users / the organization / tenant to allow indexing all drives. Since indexing is currently part of the search service we should split the two services to prevent users from gaining access to an indexer process with `Files.Read.All` application permissions by exploiting a bug in the search process.
+thumbnail | `Files.Read.All` | | The thumbnailer needs to access file content the user has access to, to generate a thumbnail. It should store them on a service storage (which is what the go micro store interface should be used for). If we want to pre generate thumbnails the service would need the `Files.Read.All` application permission, but again, a dedicated service would be preferable for security reasons.
+antivirus | | `Files.Read.All` | As part of postprocessing this is event triggered and requires an application permission.
+
+
+#### Why don't we need service accounts for services that manage and list users?
+Actually, we do. One side of introducing service accounts is giving every service a service account. The other is actually making called services check the permissions of the service account. We first need service accounts for search indexing and antivirus because they actually solve a problem there. Then we can gradually increase the security of the inter service communication by requiring more and more permissions, e.g. for managing user accounts.
+
+#### Where do we store them? In the same user backend?
+In the CS3 API we can find USER_TYPE_SERVICE and USER_TYPE_APPLICATION, where application is something like collabora and service is afaict sth like service accounts.
```

Comment:
The meaning of USER_TYPE_SERVICE  is hard to understand. Please provide a sentence to explain that.

Reply:
Moved to the updated Service Accounts section.